src/profiles/ro_crate/ro_crate_parser.py

```diff
@@ -66,15 +90,39 @@ def __init__(
         with open(CRATE_TO_TARDIS_PROFILE, encoding="utf-8") as f:
             self.mapper: CrateToTardisMapper = CrateToTardisMapper(json.load(f))
         self.crate = ROCrate(Path(crate_root_path))
-        self.crate_name = crate_name
-        if not self.crate_name:
-            # To be replaced with UUID defined in RO-Crate root
-            self.crate_name = self.crate.source.parts[-2]
-
+        self.name = crate_name
+        self.name = self._read_crate_name()
+        self.uuid = self._read_crate_uuid()
         self.filters = PathFilterSet(True)
 
+    def _read_crate_uuid(self) -> uuid.UUID:
+        root_dataset = self.crate.root_dataset
+        for identifier in as_list(root_dataset.as_jsonld().get("identifier")):
+            logger.debug("checking identifiery %s", identifier)
+            if crate_uuid := retrieve_property_value(identifier, "RO-CrateUUID"):
+                return uuid.UUID(crate_uuid)
+        logger.info(
+            "No UUID provided in RO-Crate using generated value %s from parser",
+            self.crate.uuid,
+        )
+        return uuid.UUID(self.crate.uuid)
+
+    def _read_crate_name(self) -> str:
+        if self.name:
+            return self.name
+        root_dataset = self.crate.root_dataset
+        for identifier in as_list(root_dataset.as_jsonld().get("identifier")):
+            if crate_name := retrieve_property_value(identifier, "RO-CrateName"):
+                return str(crate_name)
+        crate_bagit_dir = self.crate.source.parts[-2]
```

Comment:
Is it always guaranteed that `-2` will be a valid index here? (Sorry, my knowledge of the data structure is limited)

Reply:
it should be if the RO-Crate is properly packaged as a bagit, But as I'm not doing any bagit processing or checksum checking yet you're right I shouldn't assume that (as there's really nothing forcing the crate to have that structure right now). I'll take it out and can add something similar back in when adding bagit processing.